Tests/ReservationTests/Controller/ReservationControllerTests.cs

```diff
@@ -72,5 +77,151 @@ public void GivenFreeSlotsVM_WhenFreeSlotsIsCalled_ThenReturnsViewResult()
             _slotService.Verify(x => x.GetFreeByParkingZoneIdAndPeriod(vm.ParkingZoneId, vm.StartTime, vm.Duration), Times.Once);
         }
         #endregion
+
+        #region Reserve
+        [Fact]
+        public void GivenSlotIdAndStartTimeAndDuration_WhenReserveIsCalled_ThenReturnsNotFound()
+        {
+            //Arrange
+            int duration = 1;
+            DateTime dateTime = DateTime.Now;
+            _slotService.Setup(x => x.GetById(Id));
+
+            //Act
+            var result = _controller.Reserve(Id, dateTime, duration);
+
+            //Assert
+            Assert.NotNull(result);
+            Assert.IsType<NotFoundResult>(result);
+            _slotService.Verify(x => x.GetById(Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenSlotIdAndStartTimeAndDuration_WhenReserveIsCalled_ThenReturnsViewResult()
+        {
+            //Arrange
+            ParkingZone zone = new ();
+            ParkingSlot slot = new ();
+            slot.ParkingZone = zone;
+            int duration = 1;
+            DateTime dateTime = DateTime.Now;
+            _slotService.Setup(x => x.GetById(Id)).Returns(slot);
+
+            //Act
+            var result = _controller.Reserve(Id, dateTime, duration);
+
+            //Assert
+            Assert.NotNull(result);
+            var model = Assert.IsType<ViewResult>(result).Model;
+            Assert.IsType<ReserveVM>(model);
+            _slotService.Verify(x => x.GetById(Id), Times.Once);
+        }
+
+        [Fact]
+        public void GivenReserveViewModel_WhenReserveIsCalledToPost_ThenReturnsNotFound()
```

Comment:
`public void `GivenReserveViewModel_WhenReserveIsCalledToPost_ThenReturnsNotFound()`

better version would be: 
        `public` void `GivenReserveViewModel_WhenPostReserveCalled_ThenReturnsNotFound()`